extide/gradle/netbeans-gradle-tooling/src/main/java/org/netbeans/modules/gradle/tooling/NbProjectInfoBuilder.java

```diff
@@ -203,6 +202,28 @@ class NbProjectInfoBuilder {
 
     private static final GradleVersion GRADLE_VERSION = GradleVersion.current().getBaseVersion();
 
+    /** Jar.getClassifier() was deprecated since Gradle 5.2, removed in 8.0 */
+    private static final Method GRADLE_JAR_GET_CLASSIFIER;
+
+    /** Jar.getArchivePath() was deprecated since Gradle 5.2, removed in 8.0 */
+    private static final Method GRADLE_JAR_GET_ARCHIVEPATH;
+
+    static {
+        Method getClassifier = null;
+        Method getArchivePath = null;
+        try {
+            getClassifier = Jar.class.getMethod("getClassifier");
+            getArchivePath = Jar.class.getMethod("getArchivePath");
+        } catch (ReflectiveOperationException e) {
+            if (GRADLE_VERSION.compareTo(GradleVersion.version("8.0")) < 0) {
+                LOG.error("Did not find expected method(s) on {}", Jar.class, e);
+            }
+            // else: expected on newer versions, so stay silent
+        }
```

Comment:
i would flip this inside out:
```java
        if (GRADLE_VERSION.compareTo(GradleVersion.version("8.0")) < 0) {
            try {
                getClassifier = Jar.class.getMethod("getClassifier");
                getArchivePath = Jar.class.getMethod("getArchivePath");
            } catch (NoSuchMethodException e) {
                LOG.error("Did not find expected method(s) on {}", Jar.class, e);
            }
        }
        // null on newer versions
```
no need to swallow exceptions on newer gradle versions imo. (they would show up in JFR logs too)